enterprise/app/trends/percentile_chart.tsx

```diff
@@ -0,0 +1,80 @@
+import React from "react";
+
+import * as format from "../../../app/format/format";
+import { ResponsiveContainer, ComposedChart, CartesianGrid, XAxis, YAxis, Line, Legend, Tooltip } from "recharts";
+
+interface Props {
+  title: string;
+  data: any[];
+  extractLabel: (datum: any) => string;
+  formatHoverLabel: (datum: any) => string;
+  extractP50: (datum: any) => number;
+  extractP75: (datum: any) => number;
+  extractP90: (datum: any) => number;
+  extractP95: (datum: any) => number;
+  extractP99: (datum: any) => number;
+}
+
+const PercentilesChartTooltip = ({
```

Comment:
nit: can you move this below `PercentilesChartComponent` since it's more of an implementation detail, and so that the chart props above are declared next to the chart component?

You can write it as `function PercentilesChartTooltip(...) {` instead of using an arrow function, to make the compiler happy.